mining/scheduler_test.go

```diff
@@ -34,17 +41,83 @@ func TestMineOnce(t *testing.T) {
 	assert.True(t, ts.ToSlice()[0].StateRoot.Equals(result.NewBlock.StateRoot))
 }
 
+// Fake worker porcelain
+type fakeWorkerPorcelain struct{}
+
+func (fwp *fakeWorkerPorcelain) MinerGetWorkerAddress(ctx context.Context, minerAddr address.Address, baseKey types.TipSetKey) (address.Address, error) {
+	return minerAddr, nil
+}
+
+func (fwp *fakeWorkerPorcelain) BlockTime() time.Duration {
+	return time.Millisecond
+}
+
+// TestMineOnce10Null calls mine once off of a base tipset with a ticket that
+// will only win 10 rounds and verifies that the output has 10 tickets and a
+// +10 height.
+func TestMineOnce10Null(t *testing.T) {
```

Comment:
Set the test flags. I'm torn, but since it uses `Sleep` rather than a fake clock, it's probably an integration test.

Reply:
I'm not sure I understand why we would add a clock to the scheduler.  Right now the worker does all of the waiting around and the scheduler helplessly blocks on serial worker calls.  I'm guessing I'm missing something, please fill me in if you see this.

Reply:
We would replace this sleep call in the scheduler with a call to our own clock: https://github.com/filecoin-project/go-filecoin/blob/bcbd054bdf113a5ade5652c51a2d9249a0295983/mining/scheduler.go#L108-L109

When writing a unit test we can swap it out for a fake clock and avoid depending on time.

Reply:
Aha! I missed that sleep.  All clear thx.